Plasma Mobile Gear: 21.05 -> 21.06 -> 21.07 -> 21.08

samueldr opened the pull request:
#### Motivation for this change

 - https://www.plasma-mobile.org/2021/06/10/plasma-mobile-update-june/
 - https://www.plasma-mobile.org/info/plasma-mobile-gear-21-06/
 - https://www.plasma-mobile.org/2021/07/20/plasma-mobile-gear-21-07/
 - https://www.plasma-mobile.org/2021/08/31/plasma-mobile-gear-21-08/

#### Issues

**:heavy_check_mark:  YES ready for review now**

 - kweather **will segfault under X11**, but runs just fine under Plasma Mobile. ¯\\\_(ツ)\_/¯


#### Relevant people

cc @SCOTT-HAMILTON (for keysmith upgrade)
cc @dotlambda (for angelfish shuffling around)





#### Things done



- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I ran all the apps under X11, and under a Plasma Mobile VM.

samueldr committed:
kweather: init at 21.06

samueldr committed:
plasmaMobileGear: 21.05 -> 21.06

 - https://www.plasma-mobile.org/2021/06/10/plasma-mobile-update-june/
 - https://www.plasma-mobile.org/info/plasma-mobile-gear-21-06/

New:

 - kasts
 - keysmith
 - kweather
 - tokodon

samueldr committed:
pulseaudio-qt: 1.2.0 -> 1.3.0

samueldr committed:
kasts: init at 21.06

samueldr committed:
keysmith: 0.2 -> 21.06

samueldr commented:
As expected:

 - https://www.plasma-mobile.org/2021/07/20/plasma-mobile-gear-21-07/

SuperSandro2000 commented on pkgs/applications/plasma-mobile/koko.nix:
```diff
@@ -0,0 +1,81 @@
+    qtquickcontrols2
+  ];
+
+  prePatch = ''
```
You are sure this should not be post?

SuperSandro2000 commented on pkgs/applications/plasma-mobile/koko.nix:
```diff
@@ -0,0 +1,81 @@
+    description = "Image gallery mobile application";
+    homepage = "https://apps.kde.org/koko/";
+    # LGPL-2.1-only OR LGPL-3.0-only OR LicenseRef-KDE-Accepted-LGPL
+    license = [ licenses.lgpl3Only licenses.lgpl21Only ];
```
```suggestion
    license = with licenses; [ lgpl3Only lgpl21Only ];
```

SuperSandro2000 commented on pkgs/applications/kde/kpublictransport.nix:
```diff
@@ -0,0 +1,13 @@
+
+mkDerivation {
+  pname = "kpublictransport";
+  meta = {
+    license = [ lib.licenses.cc0 ];
+    maintainers = [ lib.maintainers.samueldr ];
+  };
```
```suggestion
  meta = with lib; {
    license = [ licenses.cc0 ];
    maintainers = [ maintainers.samueldr ];
  };
```
Also meta should be last.

samueldr committed:
kweathercore: init at v0.4

Why 0.4 when 0.5 is tagged? Because current kweather depends on 0.4 and
won't build with 0.5...

¯\_(ツ)_/¯

samueldr committed:
plasmaMobileGear: 21.06 -> 21.07

 - https://www.plasma-mobile.org/2021/07/20/plasma-mobile-gear-21-07/

(Nothing new)

samueldr committed:
ktrip: Fix build for 21.07

samueldr committed:
spacebar: Fix build for 21.07

samueldr committed:
plasmaMobileGear: 21.07 -> 21.08

 - https://www.plasma-mobile.org/2021/08/31/plasma-mobile-gear-21-08/

New:

 - audiotube

samueldr committed:
kweathercore: 0.4 -> 0.5

samueldr committed:
kweather: Fix build for 21.08

samueldr committed:
audiotube: init at 21.08

samueldr committed:
angelfish: Provide as Plasma Mobile Gear

samueldr commented:
@dotlambda review requested for angelfish

@SCOTT-HAMILTON review requested for keysmith

samueldr commented:
### TODO

 - [ ] Break evaluation of `angelfish`'s cargo dependencies to prevent accidental bad updates

samueldr committed:
angelfish: Break build on partial updates

The hardcoded cargo hash could have been left unchanged on package
updates. By breaking the build explicitly we ensure partial updates
won't accidentally slip past us.
